source/common/router/config_impl.cc

```diff
@@ -976,8 +975,8 @@ absl::optional<std::string> RouteEntryImplBase::currentUrlPathAfterRewriteWithMa
       auto just_path(Http::PathUtil::removeQueryAndFragment(path));
       return path.replace(0, just_path.size(), rewrite);
     }
-    ASSERT(case_sensitive_ ? absl::StartsWith(path, matched_path)
-                           : absl::StartsWithIgnoreCase(path, matched_path));
+    ASSERT(case_sensitive() ? absl::StartsWith(path, matched_path)
```

Comment:
`case_sensitive_` was declared `protected` in the middle of the class. Moving it down, I made it `private`. And instead of introducing another `protected` block for just one bool at the end the class, I switched subclasses to access it via a getter.